contrib/go/src/python/pants/contrib/go/tasks/go_fetch.py

```diff
@@ -47,7 +53,27 @@ def execute(self):
     if not go_remote_libs:
       return
 
-    undeclared_deps = self._transitive_download_remote_libs(set(go_remote_libs))
+    # We cache mappings from import path to root (e.g., example.org/pkg/foo -> example.org),
+    # otherwise we may have to fetch them over the network via the meta tag protocol.
+    # Note that this caching is unversioned: The mapping is defined as "whatever meta tag is
+    # currently being served at the relevant URL", which is inherently unversioned.
+    import_root_map_path = os.path.join(self.workdir, 'pkg_root_map.txt')
```

Comment:
I'd considered that, but am of two minds about it. There is a con, namely having to read and write a per-target file multiple times per target (since we recurse) instead of reading a single shared file once for all targets. And these aren't expensive to fetch once after a clean-all, although granted it would be nice not to have to do that. So I'm not sure that's worth the extra disk IO on every run?

Anyway - I've implemented it so you can see what it would look like, tell me if you think it's worth it. I could be convinced either way.